client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -233,6 +235,14 @@ export default function getThankYouPageUrl( {
 		return '/';
 	}
 
+	// Gift purchases need to bypass everything below, especially the updateUrlInCookie
+	// redirection to the ecommerce thank you page for ecommerce plan checkouts.
+	// It's possible this might need to be moved further up later if we want to cover gift purchase on Jetpack sites.
+	if ( isGiftPurchase ) {
+		debug( 'gift purchase' );
+		return `/checkout/gift/thank-you/${ siteSlug }`;
```

Comment:
I think this should work better:
```suggestion
	if ( cart?.is_gift_purchase ) {
		debug( 'gift purchase' );
		return `/checkout/gift/thank-you/${ cart?.gift_details?.receiver_blog_slug }`;
```

The reasons to do this are:
1) The `isGiftPurchase` value seems to be lost between page reloads. 
2) The `siteSlug` is always `no-site`, since the checkout works using the siteless strategy. Since we want to actually display some data about the gift receiver's site the only option is to grab that from the shopping-cart.

Note: These properties are part of D91926-code which should be deployed soon.

Reply:
Nice fix. I will say that we might _still_ want to cover the case where `is_gift_purchase` is truthy and `receiver_blog_slug` is missing since you never know what someone might change on the backend. I think this leaves us with the same three options I listed above, but I'll revise here:

1. Provide a fallback value if `receiver_blog_slug` is empty and support it in `GiftThankYou`.
1. Change the condition to also check for `receiver_blog_slug`.
1. Throw if `receiver_blog_slug` is empty but `is_gift_purchase` is set.

Reply:
Looks good, updated.

Reply:
+ Thanks for D91926-code @dylanrevisited!